sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -317,8 +320,18 @@ class JacksonParser(
 
     // SPARK-25040: Disallow empty strings for data types except for string and binary types.
     case VALUE_STRING if parser.getTextLength < 1 =>
```

Comment:
Instead of the below 14 line changes, shall we add this 7 lines before this line?
```scala
    case VALUE_STRING if parser.getTextLength < 1 && allowEmptyString =>
      dataType match {
        case FloatType | DoubleType | TimestampType | DateType =>
          throw new RuntimeException(
            s"Failed to parse an empty string for data type ${dataType.catalogString}")
        case _ => null
      }
```

Reply:
Ok. Looks good.